base/strings/search.jl

```diff
@@ -106,12 +106,10 @@ function findnext(testf::Function, s::AbstractString, i::Integer)
     z = ncodeunits(s) + 1
     1 ≤ i ≤ z || throw(BoundsError(s, i))
     @inbounds i == z || isvalid(s, i) || string_index_err(s, i)
-    while !done(s,i)
-        d, j = next(s,i)
+    for (j, d) in pairs(s[i:end])
```

Comment:
`enumerate(SubString(s, i))`

Reply:
Not the same thing. This needs string indices not ordinals.

Reply:
Should probably use a `SubString` though to avoid copying, no?

Reply:
I was under the impression that it did (because it does for AbstractString), but turns out I'm wrong for `String`, so yeah, needs to be `SubString`.